platform/os_windows.go

```diff
@@ -141,6 +142,26 @@ func (p *execClient) ExecuteCommand(command string) (string, error) {
 	cmd.Stderr = &stderr
 	cmd.Stdout = &stdout
 
+	if err := cmd.Run(); err != nil {
+		return "", errors.Wrapf(err, "ExecuteRawCommand failed. stdout: %q, stderr: %q", stdout.String(), stderr.String())
+	}
+
+	return stdout.String(), nil
+}
+
+func (p *execClient) ExecuteCommand(command string, args ...string) (string, error) {
+	if p.logger != nil {
+		p.logger.Info("[Azure-Utils]", zap.String("ExecuteCommand", command), zap.Strings("args", args))
+	} else {
+		log.Printf("[Azure-Utils] ExecuteCommand: %q %v", command, args)
+	}
+
+	var stderr, stdout bytes.Buffer
+
+	cmd := exec.Command(command, args...)
```

Comment:
Currently the execClient already takes in a timeout so we're using that timeout as the context for each command. I've added the context to the new method so that it respects the timeout unlike the deprecated ExecuteRawCommand method.

Reply:
you should still pass in a parent context and use it instead of `context.Background` when you make the scoped one so that graceful cancellation can be propagated